### PR TITLE
[DOCS-1682] Document GraphQL naming constraints for metrics

### PR DESCRIPTION
**Pull request by mdlinville**
## Summary

This PR documents GraphQL naming limitations for metrics in response to customer issue WB-26472, where metrics containing commas could not be sorted in the W&B UI.

Ready for peer review.

Previews:
- Added details: https://docs-graphql-metric-naming-l.docodile.pages.dev/guides/track/log/#metric-naming-constraints
- Added a link for more details: https://docs-graphql-metric-naming-l.docodile.pages.dev/guides/track/limits/#logging-considerations
- New KB article: https://docs-graphql-metric-naming-l.docodile.pages.dev/support/metric-naming-graphql-limitations/

Where possible, common content is shared via includes.

## Problem

Metrics with special characters (commas, spaces, hyphens, etc.) cannot be sorted or filtered in the W&B UI due to GraphQL naming constraints.

## Solution

Added comprehensive documentation in three key locations:

### 1. Main logging documentation (`content/en/guides/models/track/log/_index.md`)
- Added new "Metric naming constraints" section
- Included detailed rules about allowed characters
- Provided code examples of valid and invalid metric names

### 2. Limits documentation (`content/en/guides/models/track/limits.md`)
- Added a brief note with cross-reference to the main documentation
- Keeps the focus on performance considerations while noting the naming constraints

### 3. New KB article (`content/en/support/kb-articles/metric-naming-graphql-limitations.md`)
- Created troubleshooting article for users encountering this issue
- Clear explanation of the problem and recommended solutions
- Practical guidance to replace special characters with underscores

## Technical Details

Per GraphQL specification, field names must:
- Contain only letters (A-Z, a-z), digits (0-9), and underscores (_)
- Start with a letter or underscore
- Match pattern `/^[_a-zA-Z][_a-zA-Z0-9]*$/`

## Testing

- [ ] Verified all documentation links work correctly
- [ ] Checked for consistent messaging across all three locations
- [ ] Validated code examples are syntactically correct

## References

- DOCS-1682: Documentation ticket
- WB-26472: Original customer bug report
- [GraphQL Specification](https://spec.graphql.org/draft/)

cc: @wandb/docs-team

📄 **[View preview links for changed pages](https://github.com/wandb/docs/pull/1481#issuecomment-3192604651)**

### CODE REVIEW

File: content/en/guides/models/track/limits.md [hunk 1]
```diff
@@ -40,6 +40,8 @@ with wandb.init() as run:
 W&B automatically flattens nested values. This means that if you pass a dictionary, W&B turns it into a dot-separated name. For config values, W&B supports 3 dots in the name. For summary values, W&B supports 4 dots.
 {{% /alert %}}
 
+Note that metric names must follow certain naming constraints to ensure they can be sorted and filtered in the UI. See [Metric naming constraints]({{< relref "/guides/models/track/log/#metric-naming-constraints" >}}) for details.
```
Comment (mdlinville) [inline content/en/guides/models/track/limits.md]:
```suggestion
Metric names must follow certain naming constraints imposed by GraphQL. See [Metric naming constraints]({{< relref "/guides/models/track/log/#metric-naming-constraints" >}}) for details.
```

File: content/en/guides/models/track/log/_index.md [hunk 2]
```diff
@@ -68,10 +68,35 @@ With W&B, you can decide exactly what you want to log. The following lists some
 
 
 
+## Metric naming constraints
+
+Due to GraphQL limitations, metric names in W&B must follow specific naming rules:
+
+* **Allowed characters**: Letters (A-Z, a-z), digits (0-9), and underscores (_)
+* **Starting character**: Names must start with a letter or underscore
```
Comment (mdlinville) [inline content/en/guides/models/track/log/_index.md]: Please move this bullet point above the previous one

File: content/en/guides/models/track/log/_index.md [hunk 3]
```diff
@@ -68,10 +68,35 @@ With W&B, you can decide exactly what you want to log. The following lists some
...
+* **Pattern**: Metric names should match `/^[_a-zA-Z][_a-zA-Z0-9]*$/`
+
+{{% alert color="warning" %}}
+Metrics with invalid characters (such as commas, spaces, or special symbols) may not be sortable or queryable in the W&B UI. Consider using underscores instead of spaces or other special characters.
```
Comment (mdlinville) [inline content/en/guides/models/track/log/_index.md]:
```suggestion
Avoid naming metrics with invalid characters (such as commas, spaces, or special symbols), which may cause problems with sorting, querying, or display in the W&B UI.
```

File: content/en/guides/models/track/log/_index.md [hunk 4]
```diff
@@ -68,10 +68,35 @@ With W&B, you can decide exactly what you want to log. The following lists some
...
+# Valid metric names
+wandb.log({"accuracy": 0.9, "val_loss": 0.1, "epoch_5": 5})
+wandb.log({"modelAccuracy": 0.95, "learning_rate": 0.001})
+
+# Invalid metric names (avoid these)
+wandb.log({"acc,val": 0.9})  # Contains comma
+wandb.log({"loss-train": 0.1})  # Contains hyphen
+wandb.log({"test acc": 0.95})  # Contains space
+wandb.log({"5_fold_cv": 0.8})  # Starts with number
+```
```
Comment (mdlinville) [inline content/en/guides/models/track/log/_index.md]: Reformat this as two lists: Example metric names and Invalid metric names. Each list item should be separately marked up in code font.

File: content/en/guides/models/track/log/_index.md [hunk 5]
```diff
@@ -68,10 +68,35 @@ With W&B, you can decide exactly what you want to log. The following lists some
...
-2. **View multiple metrics on one chart**: Log multiple metrics in the same call to `wandb.Run.log()`, like this: `wandb.log({"acc'": 0.9, "loss": 0.1})` and they will both be available to plot against in the UI
+2. **View multiple metrics on one chart**: Log multiple metrics in the same call to `wandb.Run.log()`, like this: `wandb.log({"acc": 0.9, "loss": 0.1})` and they will both be available to plot against in the UI
 3. **Customize the x-axis**: Add a custom x-axis to the same log call to visualize your metrics against a different axis in the W&B dashboard. For example: `wandb.Run.log({'acc': 0.9, 'epoch': 3, 'batch': 117})`. To set the default x-axis for a given metric use [Run.define_metric()]({{< relref "/ref/python/sdk/classes/run.md#define_metric" >}})
 4. **Log rich media and charts**: `wandb.Run.log()` supports the logging of a wide variety of data types, from [media like images and videos]({{< relref "./media.md" >}}) to [tables]({{< relref "./log-tables.md" >}}) and [charts]({{< relref "/guides/models/app/features/custom-charts/" >}}).
```
Comment (mdlinville) [inline content/en/guides/models/track/log/_index.md]: Review this section to make sure that sentences are complete and punctuated.

File: content/en/support/kb-articles/metric-naming-graphql-limitations.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
+---
+url: /support/:filename
+title: "Why can't I sort or filter metrics with special characters?"
```
Comment (mdlinville) [inline content/en/support/kb-articles/metric-naming-graphql-limitations.md]:
```suggestion
title: "Why can't I sort or filter metrics with certain characters?"
```

File: content/en/support/kb-articles/metric-naming-graphql-limitations.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+toc_hide: true
+type: docs
+support:
+   - experiments
+---
+Due to GraphQL limitations, metrics containing special characters (such as commas, hyphens, spaces, or symbols) may not be sortable or filterable in the W&B UI.
```
Comment (mdlinville) [inline content/en/support/kb-articles/metric-naming-graphql-limitations.md]: Rephrase this content to be more similar to the copy near https://github.com/wandb/docs/pull/1481/files#diff-f66283c952d7d8e9198480a2d8632ac93ecacdbcce51ffb410501734d21ba80cR71, to explain what characters a metric is allowed to have. Then you can describe the problems you can run into otherwise.

File: content/en/support/kb-articles/metric-naming-graphql-limitations.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+## Valid metric names
+
+Metric names must follow these rules:
```
Comment (mdlinville) [inline content/en/support/kb-articles/metric-naming-graphql-limitations.md]: It looks like we can probably create an include with the metric naming rules, and use `readfile` to include that content in multiple places. By convention, includes are in `content/en/_includes/`.

File: content/en/support/kb-articles/metric-naming-graphql-limitations.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+- **Allowed characters**: Letters (A-Z, a-z), digits (0-9), and underscores (_)
+- **Starting character**: Must start with a letter or underscore
+- **Pattern**: Should match `/^[_a-zA-Z][_a-zA-Z0-9]*$/`
+
+## Examples
+
+**Valid metric names:**
```
Comment (mdlinville) [inline content/en/support/kb-articles/metric-naming-graphql-limitations.md]: Similarly, this could be reused, after addressing the feedback I left about it in the log/_index.md.

File: content/en/support/kb-articles/metric-naming-graphql-limitations.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+**Invalid metric names that may cause sorting issues:**
+```python
+wandb.log({"acc,val": 0.9})  # Contains comma
+wandb.log({"loss-train": 0.1})  # Contains hyphen
+wandb.log({"test acc": 0.95})  # Contains space
+```
+
+## Recommended solution
+
+Replace special characters with underscores when naming metrics:
```
Comment (mdlinville) [inline content/en/support/kb-articles/metric-naming-graphql-limitations.md]:
```suggestion
Replace invalid characters with valid characters such as underscores:
```

File: content/en/guides/models/track/log/_index.md [hunk 11]
```diff
@@ -68,10 +68,22 @@ With W&B, you can decide exactly what you want to log. The following lists some
...
+{{% /alert %}}
+
+{{< readfile file="/_includes/metric-naming-examples.md" >}}
+
 ## Common workflows
 
 1. **Compare the best accuracy**: To compare the best value of a metric across runs, set the summary value for that metric. By default, summary is set to the last value you logged for each key. This is useful in the table in the UI, where you can sort and filter runs based on their summary metrics, to help compare runs in a table or bar chart based on their _best_ accuracy, instead of final accuracy. For example: `wandb.run.summary["best_accuracy"] = best_accuracy`
-2. **View multiple metrics on one chart**: Log multiple metrics in the same call to `wandb.Run.log()`, like this: `wandb.log({"acc'": 0.9, "loss": 0.1})` and they will both be available to plot against in the UI
+2. **View multiple metrics on one chart**: Log multiple metrics in the same call to `wandb.Run.log()`, like this: `wandb.log({"acc": 0.9, "loss": 0.1})` and they will both be available to plot against in the UI
```
Comment (mdlinville) [inline content/en/guides/models/track/log/_index.md]:
```suggestion
2. **View multiple metrics on one chart**: Log multiple metrics in the same call to `wandb.Run.log()`, like this: `wandb.log({"acc": 0.9, "loss": 0.1})` and they will both be available to plot against in the UI.
```

File: content/en/guides/models/track/log/_index.md [hunk 12]
```diff
@@ -68,10 +68,22 @@ With W&B, you can decide exactly what you want to log. The following lists some
...
 3. **Customize the x-axis**: Add a custom x-axis to the same log call to visualize your metrics against a different axis in the W&B dashboard. For example: `wandb.Run.log({'acc': 0.9, 'epoch': 3, 'batch': 117})`. To set the default x-axis for a given metric use [Run.define_metric()]({{< relref "/ref/python/sdk/classes/run.md#define_metric" >}})
```
Comment (mdlinville) [inline content/en/guides/models/track/log/_index.md]:
```suggestion
3. **Customize the x-axis**: Add a custom x-axis to the same log call to visualize your metrics against a different axis in the W&B dashboard. For example: `wandb.Run.log({'acc': 0.9, 'epoch': 3, 'batch': 117})`. To set the default x-axis for a given metric use [Run.define_metric()]({{< relref "/ref/python/sdk/classes/run.md#define_metric" >}}):
```

File: content/en/guides/models/track/log/_index.md (hunk 4 above)
Comment (mdlinville) [reply content/en/guides/models/track/log/_index.md]: This feedback still needs to be addressed, but the content is now in an include.

**Review by ktmathews89 — APPROVED** [review]
this looks good to me!

File: content/en/guides/models/track/log/_index.md [hunk 13]
```diff
@@ -68,11 +68,23 @@ With W&B, you can decide exactly what you want to log. The following lists some
...
-2. **View multiple metrics on one chart**: Log multiple metrics in the same call to `wandb.Run.log()`, like this: `wandb.log({"acc'": 0.9, "loss": 0.1})` and they will both be available to plot against in the UI
-3. **Customize the x-axis**: Add a custom x-axis to the same log call to visualize your metrics against a different axis in the W&B dashboard. For example: `wandb.Run.log({'acc': 0.9, 'epoch': 3, 'batch': 117})`. To set the default x-axis for a given metric use [Run.define_metric()]({{< relref "/ref/python/sdk/classes/run.md#define_metric" >}})
+2. **View multiple metrics on one chart**: Log multiple metrics in the same call to `wandb.Run.log()`, like this: `wandb.log({"acc": 0.9, "loss": 0.1})` and they will both be available to plot against in the UI.
+3. **Customize the x-axis**: Add a custom x-axis to the same log call to visualize your metrics against a different axis in the W&B dashboard. For example: `wandb.Run.log({'acc': 0.9, 'epoch': 3, 'batch': 117})`. To set the default x-axis for a given metric use [Run.define_metric()]({{< relref "/ref/python/sdk/classes/run.md#define_metric" >}}):
```
Comment (mdlinville) [inline content/en/guides/models/track/log/_index.md]:
```suggestion
3. **Customize the x-axis**: Add a custom x-axis to the same log call to visualize your metrics against a different axis in the W&B dashboard. For example: `wandb.Run.log({'acc': 0.9, 'epoch': 3, 'batch': 117})`. To set the default x-axis for a given metric use [Run.define_metric()]({{< relref "/ref/python/sdk/classes/run.md#define_metric" >}}).
```

**Comment by mdlinville** [comment]
@dmitryduev Brenda thought you would be a good technical reviewer for these recommendations, from a SDK point of view. This is to help customers not get into the situation where sorting by metrics with commas was breaking the workspace, among other things. Thanks in advance.

**Review by dmitryduev — APPROVED** [review]
Looks great!

**Review by ngrayluna — APPROVED** [review]
